lib/dialyzer/src/dialyzer_codeserver.erl

```diff
@@ -98,9 +98,9 @@
 
 %% We KNOW that `error` is not a valid value in the table.
 ets_dict_find(Key, Table) ->
-  case ets:lookup_element(Table, Key, 2, error) of
-    error -> error;
-    Val -> {ok, Val}
+  case ets:lookup_element(Table, Key, 2, '$missing') of
```

Comment:
I'm always suspicious when actually possible values are used on the grounds that it should not be possible to have them present because of circumstances (circumstances may change), or that it is unlikely to have such a value because it is obscure.

To be safe, the default should be a reference created with `make_ref()`, which is guaranteed to be unique.

Otherwise, admitting that creating references does not come for free as pointed out when `lookup_element/4` was introduced, it may be allowed to do it this way but should be noted somewhere that the atom `'$missing'` (and `'$missing_key'` for that matter) _must not_ be inserted into the table.